README.md

```diff
@@ -222,17 +222,94 @@ CREATE TABLE users (
 
 ### EQL functions
 
-EQL provides specialized functions to interact with encrypted data:
+EQL provides specialized functions to interact with encrypted data.
+These Functions expect an encrypted domain type (which is effectively just JSONB).
 
 - **`cs_ciphertext_v1(val JSONB)`**: Extracts the ciphertext for decryption by CipherStash Proxy.
 - **`cs_match_v1(val JSONB)`**: Enables basic full-text search.
 - **`cs_unique_v1(val JSONB)`**: Retrieves the unique index for enforcing uniqueness.
 - **`cs_ore_v1(val JSONB)`**: Retrieves the Order-Revealing Encryption index for range queries.
-- **`cs_ste_vec_v1(val JSONB)`**: Retrieves the Structured Encryption Vector for containment queries.
+
+
+#### `cs_ste_vec_v1(val JSONB)`
+
+Retrieves the Structured Encryption Vector for containment queries.
+
+**Example:**
+
+```rb
+# Serialize a JSONB value bound to the users table column
+term = User::ENCRYPTED_JSONB.serialize({field: "value"})
+User.where("cs_ste_vec_v1(attrs) @> cs_ste_vec_v1(?)", term)
+```
+
+Which will execute on the server as:
+
+```sql
+SELECT * FROM users WHERE cs_ste_vec_v1(attrs) @> '53T8dtvW4HhofDp9BJnUkw';
+```
+
+And is the EQL equivalent of the following plaintext query.
+
+```sql
+SELECT * FROM users WHERE attrs @> '{"field": "value"}`;
+```
+
+#### `cs_ste_term_v1(val JSONB, epath TEXT)`
```

Comment:
Could we make `epath` here json/jsonb as well? The motivation is that the JSON value would include all the info needed for generating the MAC so that Proxy doesn't need to sort out the table, column, etc based by looking at anything other than the param or literal used for the arg.

I think that we could aim to target text in SM2, but using JSON here similar to how encryption already works would simplify the Proxy (MLP) logic.

Reply:
Also, what's the return type of `cs_ste_term_v1 `? `ore_64_8_v1`?

Reply:
Yeah, I think for now that makes the most sense. I considered adding a variant of `cs_ste_term_v1` which also takes a term type so we can handle other types of index terms but for now this should be enough. Does that sounds reasonable?